### ISSUE TITLE
CentOS http Status 500

### ISSUE THREAD

**Issue by schunt1** [OraOpenSource/OXAR#194]
Ok, so I've given up on Debian and gone with CentOS instead.  Install ran through perfectly first time, but unfortunately I get a server 500 when I try and access the web server (52.174.166.222).  I have tried shutting down and restarting tomcat, but still no cigar.  I have attached the install log; can you see if I'm missing something obvious?  

[centos-instal.txt](https://github.com/OraOpenSource/OXAR/files/513382/centos-instal.txt)
[centos-error.txt](https://github.com/OraOpenSource/OXAR/files/513384/centos-error.txt)

**Comment by martindsouza** [comment]
@tschf I think this issue is similar to previous ones (related to `tomcat@oxar`) where tomcat takes a while (still don't know why) to get everything working. 

Any ideas?

**Comment by tschf** [comment]
I think I had this once a while back, and didn't find the root cause, and ended up either re-installing Tomcat or ORDS (or both). It was just a blank white page with a 500 error i believe. fwiw, this was before the tomcat@oxar change.

Glancing at that error log, there is this (not sure if normal):

```
Oct 04, 2016 7:13:15 PM oracle.dbtools.rt.config.setup.SchemaSetup verifyDBAUserConnection
WARNING: Failed to connect to user sys AS SYSDBA jdbc:oracle:thin:@localhost:1521:xe 
ORA-00604: error occurred at recursive SQL level 1
ORA-01882: timezone region not found
```

Not very meaningful..

**Comment by GasparYYC** [comment]
I think this is not a problem anymore, as per issue [Issue #200](https://github.com/OraOpenSource/OXAR/issues/200) once the enthropy pool has sufficient random numbers the installation of OXAR (and ORDS in particular) shows no errors whatsover.
Also it's way faster than before, used to take about 15 mins, now <1 minute.
I used to get that Failed to Connect error.

Note: This is ORDS 3.0.6

This is the log (no errors):
`######################################################################## 100.0%
Oct 27, 2016 11:17:49 AM oracle.dbtools.cmdline.ModifyConfigDir execute
INFO: Set config.dir to /etc in: /root/oxar/tmp/ords/ords.war
Oct 27, 2016 11:17:59 AM oracle.dbtools.common.config.file.ConfigurationFilesBase update
INFO: Updated configurations: defaults, apex, apex_pu, apex_al, apex_rt
Oct 27, 2016 11:18:00 AM oracle.dbtools.installer.Installer log
INFO: Installing Oracle REST Data Services version 3.0.6.176.08.46
Oct 27, 2016 11:18:00 AM oracle.dbtools.installer.Runner log
INFO: ... Log file written to /root/oxar/tmp/ords/logs/ords_install_core_2016-10-27_111800_00102.log
Oct 27, 2016 11:18:04 AM oracle.dbtools.installer.Runner log
INFO: ... Verified database prerequisites
Oct 27, 2016 11:18:06 AM oracle.dbtools.installer.Runner log
INFO: ... Created Oracle REST Data Services schema
Oct 27, 2016 11:18:07 AM oracle.dbtools.installer.Runner log
INFO: ... Created Oracle REST Data Services proxy user
cat Oct 27, 2016 11:18:10 AM oracle.dbtools.installer.Runner log
INFO: ... Granted privileges to Oracle REST Data Services
Oct 27, 2016 11:18:18 AM oracle.dbtools.installer.Runner log
INFO: ... Created Oracle REST Data Services database objects
Oct 27, 2016 11:18:31 AM oracle.dbtools.installer.Runner log
INFO: ... Log file written to /root/oxar/tmp/ords/logs/ords_install_datamodel_2016-10-27_111831_00497.log
Oct 27, 2016 11:18:33 AM oracle.dbtools.installer.Installer log
INFO: Completed installation for Oracle REST Data Services version 3.0.6.176.08.46. Elapsed time: 00:00:33.139

Oct 27, 2016 11:18:36 AM oracle.dbtools.rt.config.setup.SetProperty execute
INFO: Modified: /etc/ords/defaults.xml, settting: security.verifySSL = false
Oct 27, 2016 11:18:39 AM oracle.dbtools.rt.config.setup.SetProperty execute
INFO: Modified: /etc/ords/defaults.xml, settting: security.requestValidationFunction = wwv_flow_epg_include_modules.authorize
Oct 27, 2016 11:18:42 AM oracle.dbtools.rt.config.setup.SetProperty execute
INFO: Modified: /etc/ords/defaults.xml, settting: apex.excel2collection = true
Oct 27, 2016 11:18:45 AM oracle.dbtools.rt.config.setup.SetProperty execute
INFO: Modified: /etc/ords/defaults.xml, settting: apex.excel2collection.useSheetName = true
`

Also the SQL Log for ORDS showing no errors:
`INFO: 11:18:30 Validating objects for Oracle REST Data Services.
VALIDATION: 11:18:30 Starting validation with connected user: SYS, schema:
ORDS_METADATA
VALIDATION: 11:18:31 Total objects: 192, compiled objects: 0, failed
compilation: 0
VALIDATION: 11:18:31     72 INDEX
VALIDATION: 11:18:31     1 JOB
VALIDATION: 11:18:31     8 PACKAGE
VALIDATION: 11:18:31     8 PACKAGE BODY
VALIDATION: 11:18:31     1 SEQUENCE
VALIDATION: 11:18:31     27 TABLE
VALIDATION: 11:18:31     26 TRIGGER
VALIDATION: 11:18:31     16 TYPE
VALIDATION: 11:18:31     6 TYPE BODY
VALIDATION: 11:18:31     27 VIEW

VALIDATION: 11:18:31 Validating ORDS Public Synonyms
VALIDATION: 11:18:31 Validation completed.
INFO: 11:18:31 Completed validation for Oracle REST Data Services.
`

@schunt1 Follow the recommendation on  [Issue #200](https://github.com/OraOpenSource/OXAR/issues/200) and try again.

**Comment by tschf** [comment]
I just tried a VM in Azure, and I too am getting this error. 

I verify with:

```bash
[trent@coastal-apps1 webapps]$ java -jar ords.war validate
Enter the name of the database server [localhost]:
Enter the database listen port [1521]:
Enter 1 to specify the database service name, or 2 to specify the database SID [1]:
Enter the database service name:xe
Please login with SYSDBA privileges to verify Oracle REST Data Services schema.

Enter the username with SYSDBA privileges to verify the installation [SYS]:
Enter the database password for SYS:
Confirm password:
Oct 30, 2016 7:55:16 AM oracle.dbtools.rt.config.setup.SchemaSetup verifyDBAUserConnection
WARNING: Failed to connect to user SYS AS SYSDBA jdbc:oracle:thin:@//localhost:1521/xe 
ORA-00604: error occurred at recursive SQL level 1
ORA-01882: timezone region not found

[trent@coastal-apps1 webapps]$ 
```

So must be some missing timezone info on these Azure images..

**Comment by tschf** [comment]
Just trying using the Oracle tzupdater tool, after tweaking a few issues, it made no difference.
I thought also I may be able to reproduce with sqlcl, but nope.

```
 ./sql sys/oracle@jdbc:oracle:thin:@//localhost:1521/xe as sysdba

SQLcl: Release 4.2.0 Production on Sun Oct 30 08:59:07 2016

Copyright (c) 1982, 2016, Oracle.  All rights reserved.

Connected to:
Oracle Database 11g Express Edition Release 11.2.0.2.0 - 64bit Production
```

The packages tzdata and tzdata-java are both installed.

Everything I look up involved sql developer or similar, and setting some jdbc parameter; Tried also doing `export JAVA_OPTS="-Duser.timezone=UTC" to see if there would be any impact, but nada. Anyone else have any ideas?

**Comment by tschf** [comment]
I re-configured the systems timezone as per this thread: http://unix.stackexchange.com/questions/110522/timezone-setting-in-linux

Then re ran the ORDS install:

```
java -jar ords.war
```

After that, everything was working.

**Comment by martindsouza** [comment]
Closing this as I think it's solved in #200 (entropy)
